Libraries/CO.CDP.WebApi.Foundation/ResponseMiddleware.cs

```diff
@@ -0,0 +1,74 @@
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Extensions;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace CO.CDP.WebApi.Foundation;
+
+internal class ResponseMiddleware(
+        RequestDelegate next,
+        ILogger<ResponseMiddleware> logger,
+        Dictionary<Type, (int, string)> exceptionMap)
+{
+    public async Task Invoke(HttpContext context)
+    {
+        var requestUrl = context.Request.GetDisplayUrl();
+
+        try
+        {
+            await next.Invoke(context);
+
+            var statusCode = context.Response.StatusCode;
+
+            if (statusCode >= 400 && statusCode < 500)
+            {
+                logger.LogInformation("Response status: {statusCode}, for request: {requestUrl}", statusCode, requestUrl);
+
+                await HandleResponse(context, statusCode, new ProblemDetails { Status = statusCode });
+            }
+        }
+        catch (Exception ex)
+        {
+            var statusCode = StatusCodes.Status500InternalServerError;
+            var errorCode = "GENERIC_ERROR";
+            var pd = new ProblemDetails { Status = statusCode };
+
+            if (exceptionMap.TryGetValue(ex.GetType(), out (int status, string code) error))
+            {
+                errorCode = error.code;
+                pd.Status = statusCode = error.status;
+                pd.Detail = ex.Message;
+                logger.LogInformation(ex, "Response status: {statusCode}, for request: {requestUrl}", statusCode, requestUrl);
```

Comment:
If the exception results in 50x error, we should log an error, information otherwise.

Reply:
For uncaught exception (not in this middleware), we are logging them as error.
Do you think for known Error codes(our error map) where status is 500, should we log as exception?